src/measure_theory/card_measurable_space.lean

```diff
@@ -0,0 +1,135 @@
+    calc # (↥⋃ (j : {j // j <₁ i}), generate_measurable_rec s ↑j) ^ (ω : cardinal.{u})
+      ≤ 𝔠 ^ (ω : cardinal.{u}) : power_le_power_right J
+    ... = 𝔠 : continuum_power_omega }
+end
+
+lemma cardinal_generate_measurable_rec_le {s : set (set α)} (hs : #s ≤ 𝔠) :
+  #(⋃ i, generate_measurable_rec s i) ≤ 𝔠 :=
+begin
+  apply (mk_Union_le _).trans,
+  rw [(aleph 1).mk_ord_out],
+  exact le_trans (mul_le_mul' aleph_one_le_continuum
+    (cardinal.sup_le.2 (λ i, cardinal_generate_measurable_rec_le_aux hs i))) (by simp)
+end
+
+/-- A set in the sigma-algebra generated by a set of sets `s` is constructed at some step of the
+transfinite induction defining `generate_measurable_rec`. -/
+theorem generate_measurable_subset_rec (s : set (set α)) ⦃t : set α⦄
+  (ht : generate_measurable s t) :
+  t ∈ ⋃ i, generate_measurable_rec s i :=
+begin
+  haveI : nonempty ω₁, by simp [← mk_ne_zero_iff, ne_of_gt, (aleph 1).mk_ord_out, aleph_pos 1],
+  inhabit ω₁,
+  induction ht with u hu u hu IH f hf IH,
+  { refine mem_Union.2 ⟨default, _⟩,
+    rw generate_measurable_rec,
+    simp only [hu, mem_union_eq, true_or] },
+  { refine mem_Union.2 ⟨default, _⟩,
+    rw generate_measurable_rec,
+    simp only [union_singleton, mem_union_eq, mem_insert_iff, eq_self_iff_true, true_or] },
+  { rcases mem_Union.1 IH with ⟨i, hi⟩,
+    obtain ⟨j, hj⟩ : ∃ j, i <₁ j,
+    { apply ordinal.has_succ_of_is_limit,
+      simp [ord_aleph_is_limit 1] },
+    apply mem_Union.2 ⟨j, _⟩,
+    rw generate_measurable_rec,
+    have : ∃ a, (a <₁ j) ∧ u ∈ generate_measurable_rec s a := ⟨i, hj, hi⟩,
+    simp [this] },
+  { have : ∀ n, ∃ i, f n ∈ generate_measurable_rec s i := λ n, by simpa using IH n,
+    choose I hI using this,
+    obtain ⟨j, hj⟩ : ∃ j, ∀ k, k ∈ range I → (k <₁ j),
+    { apply ordinal.lt_cof_type,
+      simp only [is_regular_aleph_one.2, mk_singleton, ordinal.type_out],
+      have : #(range I) = lift.{0} (#(range I)), by simp only [lift_uzero],
+      rw this,
+      apply mk_range_le_lift.trans_lt _,
+      simp [omega_lt_aleph_one] },
+    apply mem_Union.2 ⟨j, _⟩,
+    rw generate_measurable_rec,
+    have : ∃ (g : ℕ → (↥⋃ (i : {i // i <₁ j}), generate_measurable_rec s i.1)),
+      (⋃ (n : ℕ), ↑(g n)) = (⋃ n, f n),
+    { refine ⟨λ n, ⟨f n, _⟩, rfl⟩,
+      exact mem_Union.2 ⟨⟨I n, hj (I n) (mem_range_self _)⟩, hI n⟩ },
+    simp [this] }
+end
+
+/-- If a sigma-algebra is generated by a set of sets with cardinality at most `𝔠`, then the sigma
+algebra satisfies the same cardinality bound. -/
+theorem cardinal_generate_measurable_le {s : set (set α)} (hs : #s ≤ 𝔠) :
+  #{t | generate_measurable s t} ≤ 𝔠 :=
+(mk_subtype_le_of_subset (generate_measurable_subset_rec s)).trans
+  (cardinal_generate_measurable_rec_le hs)
+
+/-- If a sigma-algebra is generated by a set of sets with cardinality at most `𝔠`, then the sigma
+algebra satisfies the same cardinality bound. -/
+theorem cardinal_measurable_set_le {s : set (set α)} (hs : #s ≤ 𝔠) :
```

Comment:
Why are there two def-eq versions of this theorem?

Reply:
Because `generate_from` is a slightly technical device, so the first version of the theorem could be hard to discover for users (albeit it is the natural one from the point of view of the proof), while the second one is phrased in more standard and usable form.